src/hooks/api/use-fetch-wrapper.ts

```diff
@@ -15,41 +30,78 @@ export function useFetchWrapper(prependApiUrl?: string) {
   };
 
   function request(method: string) {
-    return async function (url: string, body?: object) {
-      // only include header if `body` is non-empty
-      const contentTypeHeader = body ? { 'Content-Type': 'application/json' } : undefined;
-      const resolvedUrl = prependApiUrl ? prependApiUrl + url : url;
-
-      const response = await fetch(resolvedUrl, {
-        method: method,
-        headers: { ...contentTypeHeader }, // todo: use auth header which includes JWT
-        body: body && JSON.stringify(body),
-      });
-
-      return await handleResponse(response);
+    return async function (url: string, body?: object, numRetries = 1) {
+      return _retryFetch(url, method, body, numRetries);
     };
   }
 
   ////////////////////////
   /// helper functions ///
   ////////////////////////
 
-  async function handleResponse(response: Response) {
-    const text = await response.text();
-    const data = text && JSON.parse(text);
+  async function _retryFetch(
+    url: string,
+    method: string,
+    body: object | undefined,
+    retries = 1
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ): Promise<any> {
```

Comment:
Could we type this as unknown or will that break things?

Reply:
we can and that would be the best thing forward, it just means we need to create object schemas for the response we get back on API calls

might be worth for a future story, because being honest, the API is changing a lot

Reply:
oh, I forgot about that. Yeah, nevermind then, haha